controllers/front/CartController.php

```diff
@@ -125,7 +125,7 @@ public function displayAjaxUpdate()
 
         $productsInCart = $this->context->cart->getProducts();
         $updatedProducts = array_filter($productsInCart, array($this, 'productInCartMatchesCriteria'));
-        list(, $updatedProduct) = each($updatedProducts);
+        $updatedProduct = reset($updatedProducts);
```

Comment:
Are we sure that this returns the same result?

In the previous code, we get the second element of the array, while from now we get the first element.

What about
```suggestion
        list(, $updatedProduct) = $updatedProducts;
```

Reply:
well actually `reset` returns the key at the index 0, and the value at the index 1
so what you suggest will indeed select the second value instead of the first one, which is not the expected behavior

Reply:
`reset` does not return the key, but the value:
> Returns the value of the first array element, or FALSE if the array is empty.

Reply:
my bad I meant `each` in the original code
`reset` does return the value which ends up as the same result than the original `each` code